equinox/nn/_attention.py

```diff
@@ -256,6 +325,9 @@ def __call__(
             be a JAX array of shape `(query_seq_length, kv_seq_length)`, or (for custom
```

Comment:
The `mask="causal"` option needs documenting.

I suspect this might end up being quite long. Maybe we should pull it out into an admonition (the `!!! foo` things you see in the docs). As a reference point, se how we handle the discussion on padding for `eqx.nn.Conv`.